### ISSUE TITLE
Conflicts with Google Analytics

### ISSUE THREAD

**Issue by jobzky** [meleze/Meleze.Web#8]
I got issue with google analytics. Using Firebug on Firefox I got error:

"missing : in conditional expression
[Break On This Error]  
... var s = document.getElementsByTagName('script')[0]; s.parentNode.inse...
jquery....min.js (line 28, col 345)"

 Google analytics snippet
...
 (function () {
        var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
        ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
        var s = document.getElementsByTagName('script')[0];
        s.parentNode.insertBefore(ga, s);
    })(); 
...

The error above causes other scripts not to function. I tried not to use Meleze.Web and the error is gone and page is not compressed anymore.

**Comment by mdg1019** [comment]
I'm having the same issue. From what I can tell Meleze.Web is not processing javascript comments correctly. 

If you change the "//" in the urls to "\x2f\x2f" it'll work. The minifier is incorrectly identifying these as a comment and stripping them remainder of the line. 

Conversely, Meleze is leaving legitimate comments intact so when you use aggressive minification you get javascript syntax errors, because the browser thinks the rest of the line is a comment.

So, Meleze seems to think "//" inside of a quotes string is a comment, but doesn't think a comment is a comment.

**Comment by jobzky** [comment]
Hi thanks for the response. I keep on waiting for someone to reply and finally you did. Let me try once again your solution and give a feed back soon.

**Comment by strongeye** [comment]
Changing cut and paste Google code is a hack that I didn't want to do, so I made a local modification to MinifyJavascriptComments that makes the distinction between inline js comments (which are "//") and inline js containing "http://" or "https://" . This may not solve all cases but at least it fixes the google analytics conflict.

```
    /// <summary>
    /// Removes the single line comments from the Javascript code.
    /// (because when line returns are minimized, these comments commant the whole script).
    /// </summary>
    /// <param name="content"></param>
    /// <param name="builder"></param>
    /// <returns></returns>
    private static string MinifyJavascriptComments(string content, StringBuilder builder)
    {
        builder.Clear();
        var iscriptstart = content.IndexOf("<script");
        while (iscriptstart >= 0)
        {
            var iscriptautoend = content.IndexOf("/>", iscriptstart + 7);
            var iscriptend = content.IndexOf("</script>", iscriptstart + 7);
            if ((iscriptend < 0) || ((iscriptautoend > 0) && (iscriptautoend < iscriptend)))
            {
                break;
            }

            // We have some javascript code inside the tag
            // => we can ask a JS minifier to simplify it
            var istartcode = content.IndexOf('>', iscriptstart) + 1;
            var iendcode = iscriptend;
            var code = content.Substring(istartcode, iendcode - istartcode);
            builder.Append(content, 0, istartcode);

            if (!string.IsNullOrWhiteSpace(code))
            {
                // We remove all // comments that cause problems when minifying the HTML later
                var lines = code.Split('\n');
                foreach (var line in lines)
                {
                    var minifiedLine = line;
                    var icomment = line.IndexOf("//");
                    var doMinifyInlineComment = icomment < 5;

                    if (icomment >= 0)
                    {
                        if (!doMinifyInlineComment)
                        {
                            var shift = icomment == 5 ? 5 : 6;
                            var subline = line.Substring(icomment - shift, shift);
                            var httpIndex = subline.IndexOf("http");
                            if (httpIndex < 0 ||
                                !(subline.Length == 5 || subline.Length == 6))
                                doMinifyInlineComment = true;
                        }

                        if (doMinifyInlineComment)
                            minifiedLine = line.Substring(0, icomment).Trim();
                    }

                    if (!string.IsNullOrEmpty(minifiedLine))
                        builder.AppendLine(minifiedLine);

                }
            }

            iscriptstart = builder.Length;

            builder.Append(content, iscriptend, content.Length - iscriptend);
            content = builder.ToString();
            builder.Clear();

            iscriptstart = content.IndexOf("<script", iscriptstart);
        }
        return content;
    }
```

**Comment by meleze** [comment]
This is a duplicate of issue #14 which now fixed in 1.4.4.
// in the javascript was always considered as a comment. Now, // in strings are not removed anymore.
